src/config.c

```diff
@@ -803,15 +804,29 @@ void configGetCommand(client *c) {
     serverAssertWithInfo(c,o,sdsEncodedObject(o));
 
     for (standardConfig *config = configs; config->name != NULL; config++) {
-        if (stringmatch(pattern,config->name,1)) {
-            addReplyBulkCString(c,config->name);
-            addReplyBulkSds(c, config->interface.get(config->data));
-            matches++;
-        }
-        if (config->alias && stringmatch(pattern,config->alias,1)) {
-            addReplyBulkCString(c,config->alias);
-            addReplyBulkSds(c, config->interface.get(config->data));
-            matches++;
+        /* Hidden configs require an exact match (not a pattern) */
+        if (config->flags & HIDDEN_CONFIG) {
+            if (!strcasecmp(pattern, config->name)) {
+                addReplyBulkCString(c, config->name);
+                addReplyBulkSds(c, config->interface.get(config->data));
+                matches++;
+                break;
+            } else if (config->alias && strcasecmp(pattern, config->alias)) {
+                addReplyBulkCString(c, config->alias);
+                addReplyBulkSds(c, config->interface.get(config->data));
+                matches++;
+                break;
+            }
+        } else {
```

Comment:
i'd prefer a `continue` rather than indent the existing code.

Reply:
:+1: